library/alloc/src/vec.rs

```diff
@@ -194,6 +194,22 @@ use crate::raw_vec::RawVec;
 /// can be slow. For this reason, it is recommended to use [`Vec::with_capacity`]
 /// whenever possible to specify how big the vector is expected to get.
 ///
+/// In summary, a vector containing 1, 2 with capacity 4 can be visualized as:
+/// ```text
+/// Stack      ptr   capacity    len
```

Comment:
"Stack" is not correct. Those three fields don't have to live on the stack, e.g. `Box<Vec<T>>`.

Reply:
A vec can also live inside some other struct that lives on the heap, e.g. an `Arc`. Maybe just call it "struct"?

Reply:
But doesn't `struct` and `heap` when used together feels weird?

Reply:
Maybe just remove "Stack". We simply don't know where those three values live. We just know that the lower box lives on the heap.

Reply:
How about `Stack/Heap`? Since it could be either stack or heap.

Reply:
Try 'Layout' instead, maybe?

Reply:
> Try 'Layout' instead, maybe?

Layout isn't clear where it will be.

> Maybe something like "The Vec object itself" and "The data on the heap".

Too long. Also having "object" word itself can confuse a lot of people. "The data on the heap" is incorrect, since it could be in the stack too.